src/server/pkg/storage/chunk/storage.go

```diff
@@ -36,18 +39,15 @@ func NewStorage(objC obj.Client, opts ...StorageOption) *Storage {
 	return s
 }
 
-// NewReader creates an io.ReadCloser for a chunk.
-// (bryce) The whole chunk is in-memory right now. Could be a problem with
-// concurrency, particularly the merge process.
-// May want to handle concurrency here (pass in multiple data refs)
-func (s *Storage) NewReader(ctx context.Context, f ...ReaderFunc) *Reader {
-	return newReader(ctx, s.objC, f...)
+// NewReader creates a new Reader.
+func (s *Storage) NewReader(ctx context.Context, dataRefs ...*DataRef) *Reader {
+	return newReader(ctx, s.objC, dataRefs...)
 }
 
-// NewWriter creates an io.WriteCloser for a stream of bytes to be chunked.
+// NewWriter creates a new Writer for a stream of bytes to be chunked.
 // Chunks are created based on the content, then hashed and deduplicated/uploaded to
 // object storage.
-// The callback arguments are the chunk hash and content.
+// The callback arguments are the chunk hash and annotations.
 func (s *Storage) NewWriter(ctx context.Context, averageBits int, f WriterFunc, seed int64) *Writer {
```

Comment:
Small nit, when functions take callbacks I generally like to have the callback be the last parameter, otherwise when you call it with an inlined function the arguments after the callback can get really far away from the callsite.

Reply:
Yeah, I normally do that as well with the only exception being optional arguments. Done.